server/channels/app/bot.go

```diff
@@ -124,7 +124,9 @@ func (a *App) CreateBot(rctx request.CTX, bot *model.Bot) (*model.Bot, *model.Ap
 
 	savedBot, nErr := a.Srv().Store().Bot().Save(bot)
 	if nErr != nil {
-		a.Srv().Store().User().PermanentDelete(rctx, bot.UserId)
+		if err := a.Srv().Store().User().PermanentDelete(rctx, bot.UserId); err != nil {
+			return nil, model.NewAppError("CreateBot", "app.bot.createbot.internal_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		}
```

Comment:
I'm concerned that if both `Bot().Save` and `User().PermanentDelete` fail, the error returned by `User().PermanentDelete` is neither logged nor returned.

I suggest that we log the error from `User().PermanentDelete` instead of returning it.

Reply:
Yeah, make sense. I was bit confused about this one